test/performance_test.go

```diff
@@ -0,0 +1,153 @@
+// +build performance
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/flowcontrol"
+	knativetest "knative.dev/pkg/test"
+)
+
+const (
+	qps               = 1
+	burst             = qps
+	injectionDuration = 1 * time.Minute
+)
+
+func TestInject(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	t.Helper()
+
+	cfg := clientConfig(t)
+	kubeClient, cs := clientSets(t, cfg)
+
+	rateLimitedCfg := clientConfig(t)
+	rateLimitedCfg.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(qps, burst)
+	rateLimitedKubeClient, rateLimitedCS := clientSets(t, rateLimitedCfg)
+
+	wg := sync.WaitGroup{}
```

Comment:
I know there's some ongoing discussion about what exactly to benchmark, but I recently discovered `testing.B` and I'm wondering if it could be helpful here? It looks like some of timing + waitgroup logic could be replaced with `b.runParallel`.

Reply:
Ah I though @guillaumerose used `testing.B`.., but turns out not. Yes, maybe using it might make sense indeed.